src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -866,7 +866,7 @@ internal static TimeSpan GetDateTimeNowUtcOffsetFromUtc(DateTime time, out bool
                 index++;
             }
 
-            if (index == 0)
+            if (rulesList.Count == 0 && index < dts.Length)
```

Comment:
>rulesList.Count == 0 && index < dts.Length [](start = 16, length = 42)

I am not sure why we need to change this check at all.

Reply:
when we skip first rule the index will not be 0 and we will go the other path (we will use indices we have skipped). If we keep it this way then the first valid time zone will start in the min possible time